docs(modeling-guidance): reframe toolContainer as retroactive and document reapply fix (#9540)
camunda/camunda-docs: docs/components/modeler/reference/modeling-guidance/rules/_declaring-agentic-subprocess.md

## Declare a sub-process as agentic

This rule applies only within an ad-hoc sub-process recognized as a tool container. An ad-hoc sub-process is recognized as a tool container in either of the following ways:

> - Its `zeebe:modelerTemplate` attribute is set to `io.camunda.connectors.agenticai.aiagent.jobworker.v1`, which identifies the AI Agent job worker template. Any version of this template is supported.

Reviewer: when https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/260 merges, the `zeebe:taskDefinition` may also be io.camunda.agenticai:aiagent-job-worker:1

Author: @HanselIdes we can wait

Reviewer: https://github.com/camunda/camunda-hub/pull/27268 has the bugfix

Author: Added the taskDefinition-based detection now that camunda-hub#27268 merged. See 255b82b9f.

> - Its `zeebe:modelerTemplate` attribute is set to `io.camunda.connectors.agenticai.aiagent.jobworker.v1`, which identifies the AI Agent job worker template. Any version of this template is supported.

Reviewer 2: 🔧 "job worker" is the technical name which we are currently trying to get rid of with the new element templates - as this never surfaces to the user besides the type, let's name it `AI Agent Sub-Process` here.

```suggestion
- Its `zeebe:modelerTemplate` attribute is set to `io.camunda.connectors.agenticai.aiagent.jobworker.v1`, which identifies the AI Agent Sub-Process template. Any version of this template is supported.
```

> - It has a `zeebe:property` named `io.camunda.agenticai.toolContainer` with the value `true`, regardless of whether its tools are invoked by an AI Agent task in the same process or in a separate process. This property is the supported long-term approach.

Reviewer 2: ```suggestion
- It has a `zeebe:property` named `io.camunda.agenticai.toolContainer` with the value `true`, regardless of whether its tools are invoked by an AI Agent Task in the same process or in a separate process. This property is the supported long-term approach.
```

Every Camunda-provided AI Agent element template sets the `io.camunda.agenticai.toolContainer` property.
The property is available across all template versions. Templates declare it as a hidden property, so it never appears as a control in the properties panel.

An element template writes its properties into the diagram only when you apply it, so an ad-hoc sub-process you modeled before the property was added keeps its original XML. Existing processes need to be updated.

### Update an existing process

> While an element template is applied, the properties panel shows only the sections the template defines, and the **Extension properties** section isn't available. Instead of adding the property by hand, unlink and reapply the template:

Reviewer 2: One nuance: when there is an element template update available, updating to the latest version should also handle this without the need to unlink & re-apply.

Author: Added

## Declare a sub-process as agentic

This rule applies only within an ad-hoc sub-process recognized as a tool container. An ad-hoc sub-process is recognized as a tool container in any of the following ways:

- Its `zeebe:modelerTemplate` attribute is set to `io.camunda.connectors.agenticai.aiagent.jobworker.v1`, which identifies the AI Agent Sub-Process template. Any version of this template is supported.
- It has a `zeebe:property` named `io.camunda.agenticai.toolContainer` with the value `true`, regardless of whether its tools are invoked by an AI Agent Task in the same process or in a separate process. This property is the supported long-term approach.

> - Its `zeebe:taskDefinition` type starts with `io.camunda.agenticai:aiagent-job-worker:`, which is the type assigned by the AI Agent Sub-Process template. This covers custom element templates that use a different `zeebe:modelerTemplate` ID but assign the same type.

Reviewer: ```suggestion
- Its `zeebe:taskDefinition` type starts with `io.camunda.agenticai:aiagent-job-worker:`, which is the type assigned by the AI Agent Sub-Process template. This covers custom element templates that use a different `zeebe:modelerTemplate` ID but assign the same type, for example an agent template you publish to your organization.
```

Reviewer: @copilot commit this change and backport it to previous versions
